src/cmd-cloud-prune

```diff
@@ -0,0 +1,291 @@
+    for action in ['cloud-uploads', 'images', 'build']:
+        if action not in policy[stream]:
+            continue
+        duration = get_period_in_months(policy[stream][action])
+        ref_date = datetime.datetime.now() - relativedelta(months=int(duration))
+
+        print(f"Pruning resources of type {action} older than {duration} months ({ref_date.date()}) on stream {stream}")
+        # Enumerating in reverse to go from the oldest build to the newest one
+        for index, build in enumerate(reversed(builds_json_data["builds"])):
+            build_id = build["id"]
+            if action in build.get("policy-cleanup", []):
+                print(f"Build {build_id} has already had {action} pruning completed")
+                continue
+            build_date = parse_fcos_version(build_id)
+
+            if build_date >= ref_date:
+                break
+            for arch in build["arches"]:
+                meta_prefix = os.path.join(prefix, f"builds/{build_id}/{arch}/meta.json")
+                meta_json = get_json_from_s3(s3_client, bucket, meta_prefix)
+                # Make sure the meta.json doesn't contain any cloud_platform that is not supported for pruning yet.
+                images = get_supported_images(meta_json, unsupported, supported)
+                current_build = Build(id=build_id, images=images, arch=arch)
+
+                match action:
+                    case "cloud-uploads":
+                        prune_cloud_uploads(current_build, cloud_config, build, builds_json_data, index, args.dry_run)
+                    case "build":
+                        raise NotImplementedError
+                        # print(f"Deleting key {prefix}{build.id} from bucket {bucket}")
+                        # Delete the build's directory in S3
+                        # S3().delete_object(args.bucket, f"{args.prefix}{str(current_build.id)}")
+
+    if not args.dry_run:
+        # Save the updated builds.json to local builds/builds.json
+        save_builds_json(builds_json_data)
+
+    if args.upload_builds_json:
+        # This copies the local builds.json and updates the S3 bucket version.
+        handle_upload_builds_json(s3_client, bucket, prefix, args.dry_run, args.acl)
+
+
+def prune_cloud_uploads(current_build, cloud_config, build, builds_json_data, index, dry_run):
+    delete_cloud_resources(current_build, cloud_config, dry_run)
+    if not dry_run:
+        build.setdefault("policy-cleanup", []).append("cloud-uploads")
+        builds_json_data["builds"][index] = build
+        builds_json_data["timestamp"] = datetime.datetime.now(pytz.utc).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+def parse_s3_url(builds_source_data_url):
+    if builds_source_data_url.startswith("s3://"):
+        bucket, prefix = builds_source_data_url.split('//', 1)[1].split('/', 1)
+        prefix = prefix[:-len("/builds")]
+        return bucket, prefix
+    else:
+        raise Exception("Invalid scheme: only s3:// supported")
```

Comment:
This would be a good fit for `urllib.parse` instead: https://docs.python.org/3/library/urllib.parse.html